### ISSUE TITLE
New rule: no-empty-component-block

### ISSUE THREAD

**Issue by manniL** [vuejs/eslint-plugin-vue#952]
**Please describe what the rule should do:**
The rule should avoid empty `<style scoped></style>`, script or template tags. If there are no style rules, a style tag is not necessary.

Edge case: Tags that have a src attribute (should be valid).

**What category should the rule belong to?**
- [x] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<template>
  <h1>Hey</h1>
</template>

<script></script>

<style scoped></style>
```

```vue
<template>
  <h1>Hey</h1>
</template>

<style scoped></style>
```

```vue
<template>
</template>

<script>
export default { 
  render (h) { return h('div','Hi') }  // Or so
}
</script>

```

**Comment by daerodriguezme** [comment]
Do you know if this has any performance impact?

**Comment by manniL** [comment]
> Do you know if this has any performance impact?

No, I don't 🙈

**Comment by tyankatsu0105** [comment]
@ota-meshi 
How about refactoring `no-empty-pattern` or adding an option to the rule?

**Comment by ota-meshi** [comment]
Hi @tyankatsu0105 
I think it's good to add a new rule `vue/no-empty-component-block`.

Probably with some attributes it should be ignored. e.g. `<style src="...">`

https://vue-loader.vuejs.org/spec.html#src-imports

**Comment by tjx666** [comment]
Why this rule can't get auto fix? Any unsafe situation?

**Comment by FloEdelmann** [comment]
Please open a new issue to discuss that.
